clients/src/main/java/org/apache/kafka/common/network/SSLTransportLayer.java

```diff
@@ -600,7 +601,7 @@ public Principal peerPrincipal() throws IOException {
         try {
             return sslEngine.getSession().getPeerPrincipal();
         } catch (SSLPeerUnverifiedException se) {
-            throw new IOException(String.format("Unable to retrieve getPeerPrincipal due to %s", se));
```

Comment:
yeah, since we are caching the Principal in the authenticator, warning here makes sense.